File: Python3/Easy/WrongAnswer/number-of-rectangles-that-can-form-the-largest-square_3.py
class Solution:
    def countGoodRectangles(self, rectangles: List[List[int]]) -> int:
        
        return len([x for x in rectangles if x[0] == max(rectangles, key = lambda x: x[0])[0] or x[1] == max(rectangles, key = lambda x: x[1])[1]])

File: Rust/Easy/Accepted/number-of-rectangles-that-can-form-the-largest-square_3.rs
impl Solution {
    pub fn count_good_rectangles(rectangles: Vec<Vec<i32>>) -> i32 {
        
        let mut max = 0;
        let mut count = 0;
        for r in rectangles {
            let m = r[0].min(r[1]);
            if m > max {
                max = m;
                count = 1;
            } else if m == max {
                count += 1;
            }
        }
        count
    }
}